File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    map<int,int> m;
    void travel(TreeNode* root,int l){
        if(!root)
            return;
        m[l]+=root->val;
        travel(root->left,l+1);
        travel(root->right,l+1);
    }
        
    int maxLevelSum(TreeNode* root) {
        travel(root,1);
        int s=INT_MIN,l=-1;
        for (const auto& [idx, sum] : m) {
            if(s<sum){
                s=sum;
                l=idx;
            }
        }
        return l;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import defaultdict

class Solution:
    def maxLevelSum(self, root: TreeNode) -> int:
        sum = defaultdict(int)
        def traverse(root, lvl):
            if root:
                sum[lvl] += root.val
                traverse(root.left,  lvl+1)
                traverse(root.right, lvl+1)
			
        traverse(root, 1)
        return max(sum, key=sum.get)
        